corehq/util/log.py

```diff
@@ -113,26 +113,30 @@ def format_details(self, details):
             formatted = '\n'.join('{item[0]}: {item[1]}'.format(item=item) for item in details.items())
             return 'Details:\n{}'.format(formatted)
 
-    def get_code(self, extracted_tb):
-        trace = next((trace for trace in extracted_tb if 'site-packages' not in trace[0]), None)
-        if not trace:
-            return None
-
-        filename = trace[0]
-        lineno = trace[1]
-        offset = 10
-        with open(filename) as f:
-            code_context = list(islice(f, lineno - offset, lineno + offset))
-
-        return highlight(''.join(code_context),
-            PythonLexer(),
-            HtmlFormatter(
-                noclasses=True,
-                linenos='table',
-                hl_lines=[offset, offset],
-                linenostart=(lineno - offset + 1),
-        )
-        )
+    @staticmethod
+    def get_code(extracted_tb):
+        try:
+            trace = next((trace for trace in extracted_tb if 'site-packages' not in trace[0]), None)
+            if not trace:
+                return None
+
+            filename = trace[0]
+            lineno = trace[1]
+            offset = 10
+            with open(filename) as f:
+                code_context = list(islice(f, lineno - offset, lineno + offset))
+
+            return highlight(''.join(code_context),
+                PythonLexer(),
+                HtmlFormatter(
+                    noclasses=True,
+                    linenos='table',
+                    hl_lines=[offset, offset],
+                    linenostart=(lineno - offset + 1),
+                )
```

Comment:
haha i think i dealt with a similar issue when first writing this. agreed i'm not always super fond of dimagimon's multiline suggestions